From 2160e7e0b7842d4ef49fdd435b4a7f127d479b90 Mon Sep 17 00:00:00 2001
From: Jakub Adam <jakub.adam@jollamobile.com>
Date: Wed, 4 Jun 2014 14:48:02 +0200
Subject: [PATCH] Use correct signal name when disconnecting "NameOwnerChanged"

A disconnectSignal() call with a wrong signal name caused that hook
wasn't found and thus kept in QDBusConnectionPrivate::signalHooks
forever.

Change-Id: Id7cda225be7580529fc835b377636226abb229f9
---
 src/dbus/qdbusintegrator.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- qtbase/src/dbus/qdbusintegrator.cpp
+++ qtbase/src/dbus/qdbusintegrator.cpp
@@ -2266,7 +2266,7 @@
                     watchedServices.erase(sit);
                     disconnectSignal(dbusServiceString(), QString(), dbusInterfaceString(),
                                   QLatin1String("NameOwnerChanged"), QStringList() << hook.service, QString(),
-                                  this, SLOT(_q_serviceOwnerChanged(QString,QString,QString)));
+                                  this, SLOT(serviceOwnerChangedNoLock(QString,QString,QString)));
                 }
             }
         }
